pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -68,6 +68,22 @@
   private static class UnsupportedProtocolException extends IOException {
   }
 
+
+  private ISSPIClient createSSPI(PGStream pgStream,
+      String spnServiceClass,
+      boolean enableNegotiate,
+      Logger logger) {
+    Class[] cArg = new Class[]{PGStream.class, String.class, boolean.class, Logger.class};
+    Class c = null;
+    try {
+      c = Class.forName("org.postgresql.sspi.SSPIClient");
+      return (ISSPIClient) c.getDeclaredConstructor(cArg).newInstance(pgStream, spnServiceClass, enableNegotiate, logger);
+    } catch (Exception e) {
+      // This catched quite a lot exceptions, but until Java 7 there is no ReflectiveOperationException
+      throw new UnsupportedOperationException("You are using jar from Linux distribution or class SPPIClient cannot be loaded");
```

Comment:
This was not good. Original exception was not included into the new one, so user would have no way to debug the failure.